pyqtorch/core/utils.py

```diff
@@ -20,6 +20,8 @@
 import torch
 from numpy.typing import NDArray
 
+torch.set_default_dtype(torch.float64)
```

Comment:
maybe it would be enough to set it in circuit.py?  it would enforce it an all submodules used in the file as far as i understand?

Reply:
Let me test :)

Reply:
Doesn't seem to work unfortunately. I believe you need the line in every file where a `torch.tensor()` is initialized.

Reply:
ok thank you for testing this :) hm, so we just put it on top of every file (which wouldnt be a lot), and then also externally passed thetas will automatically converted to this?

Reply:
Well for now we can just keep it in these two files, which seems to cover all the gates being created. But yes, this fixes the thetas.

If we always include it by default in a new file we could in theory be more relaxed with the `dtype=torch.cdouble` used everywhere, as torch should initialize everything with `float64` and automatically convert to `complex128` when a complex number is detected, but I did not test this. @madagra also mentioned we could at some point have this be a global setting we could control, as we could then choose to run a given simulation in `float32 / complex64` to save memory if needed, but this would need to be properly tested in a new MR.

Reply:
Wouldn't be better to put it in the `__init__.py` of the repo? In this way, we are sure it is always called no matter what the user does. And yes, having it as a sort of tweakable configuration would be nice (see for instance [this library](https://github.com/lululxvi/deepxde/blob/master/deepxde/config.py) for a possible implementation: ). For now, let's leave it like this, though.

Reply:
I tested now to move it to the `__init__.py` of the repo but it's not producing the same thing, the results still come out as single precision.

Reply:
OK, interesting. Did you put in the `__init__.py` of the full repo, right? I mean `pyqtorch/__init__.py` and not the one of a submodule. In any case, if it does not work, let's keep it as it is and merge this one.

Reply:
Yes @madagra I tried in `pyqtorch/__init__.py`